src/components/drawer/drawer.test.tsx

```diff
@@ -0,0 +1,480 @@
+
+    const button = screen.getByRole("button", { name: "toggle sidebar" });
+    await user.click(button);
+
+    expect(await screen.findByTestId("drawer-content")).toHaveClass("opening");
+
+    act(() => {
+      jest.advanceTimersByTime(500);
+    });
+
+    const content = screen.getByTestId("drawer-content");
+    expect(content).toHaveClass("open");
+    expect(content).not.toHaveClass("opening");
+  });
+
+  it("triggers closing animation sequence with correct timing, when animationDuration prop is specified in seconds", async () => {
+    const user = userEvent.setup({
+      advanceTimers: jest.advanceTimersByTime,
+    });
+    render(
+      <Drawer
+        expanded
+        sidebar="Sidebar content"
+        showControls
+        animationDuration="0.5s"
+      >
+        Foobar
+      </Drawer>
+    );
+
+    const button = screen.getByRole("button", { name: "toggle sidebar" });
+    await user.click(button);
+
+    expect(await screen.findByTestId("drawer-content")).toHaveClass("closing");
+
+    act(() => {
+      jest.advanceTimersByTime(500);
+    });
+
+    const content = screen.getByTestId("drawer-content");
+    expect(content).toHaveClass("closed");
+    expect(content).not.toHaveClass("closing");
+  });
+
+  it("triggers opening animation sequence, when expanded prop is updated to true", async () => {
+    const MockApp = () => {
+      const [expanded, setExpanded] = React.useState(false);
+      return (
+        <>
+          <button type="button" onClick={() => setExpanded(true)}>
+            Expand
+          </button>
+          <Drawer expanded={expanded} animationDuration="500">
+            Foobar
+          </Drawer>
+        </>
+      );
+    };
+    const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+    render(<MockApp />);
+
+    await user.click(screen.getByRole("button", { name: "Expand" }));
+
+    expect(await screen.findByTestId("drawer-content")).toHaveClass("opening");
+
+    act(() => {
+      jest.advanceTimersByTime(500);
+    });
+
+    const content = screen.getByTestId("drawer-content");
+    expect(content).toHaveClass("open");
+    expect(content).not.toHaveClass("opening");
+  });
+
+  it("triggers closing animation sequence, when expanded prop is updated to false", async () => {
+    const MockApp = () => {
+      const [expanded, setExpanded] = React.useState(true);
+      return (
+        <>
+          <button type="button" onClick={() => setExpanded(false)}>
+            Collapse
+          </button>
+          <Drawer expanded={expanded} animationDuration="500">
+            Foobar
+          </Drawer>
+        </>
+      );
+    };
+    const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+    render(<MockApp />);
+
+    await user.click(screen.getByRole("button", { name: "Collapse" }));
+
+    expect(await screen.findByTestId("drawer-content")).toHaveClass("closing");
+
+    act(() => {
+      jest.advanceTimersByTime(500);
+    });
```

Comment:
Since React performs numerous state updates in the background, verifying the wrapper's correct class before fast-forwarding the animation ensures we fast-forward at the correct time.